CSharpFunctionalExtensions/Maybe/MaybeExtensions.cs

```diff
@@ -100,6 +107,48 @@ public static void Execute<T>(this Maybe<T> maybe, Action<T> action)
             action(maybe.Value);
         }
 
+        /// <summary>
+        /// Executes the given <paramref name="action" /> if the <paramref name="maybe" /> has no value
+        /// </summary>
+        /// <param name="maybe"></param>
+        /// <param name="action"></param>
+        /// <typeparam name="T"></typeparam>
+        public static void ExecuteNoValue<T>(this Maybe<T> maybe, Action action)
+        {
+            if (maybe.HasValue)
+                return;
+
+            action();
+        }
+        
+        /// <summary>
+        /// Executes the given async <paramref name="action" /> if the <paramref name="maybe" /> has a value
+        /// </summary>
+        /// <param name="maybe"></param>
+        /// <param name="action"></param>
+        /// <typeparam name="T"></typeparam>
+        public static async Task Execute<T>(this Maybe<T> maybe, Func<T, Task> action)
```

Comment:
While these are async methods they work with 'sync' `Maybe`s, so I put them in this file instead of `AsyncMaybeExtensions`. `.Match` is in here even though it could return a `Task`.

But, I can move them if you think that makes sense.

Reply:
No need, looks good as-is